backend/src/test/java/com/google/sps/TestEmailNotifier.java

```diff
@@ -0,0 +1,109 @@
+package com.google.sps;
+
+import com.google.sps.notifs.EmailNotifier;
+import org.junit.Assert;
+import org.junit.Test;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.util.Properties;
+import static org.junit.Assert.*;
+
+public class TestEmailNotifier {
+
+  public MimeMessage createTestMessage() {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    return new MimeMessage(session);
+  }
+
+  private String getTextFromMessage(MimeMessage message) throws MessagingException, IOException {
+    String result = "";
+    result = message.getContent().toString();
+    return result;
+  }
+
+  @Test
+  public void constructorShouldSetNameAndEmailTo() {
+    EmailNotifier emailNotifier = new EmailNotifier("John", "jdoe@gmail.com", null);
+    assertEquals("John", emailNotifier.getRecipientName());
+    assertEquals("jdoe@gmail.com", emailNotifier.getToEmail());
+  }
+
+  @Test
+  public void testMessageHasCorrectApplicationName() throws MessagingException {
+    EmailNotifier emailNotifier = new EmailNotifier("Jordan", "jjgrant@buffalo.edu", null);
+    MimeMessage mimeMessage = emailNotifier.createEmail("", "", "");
+    MimeMessage message = createTestMessage();
```

Comment:
Similar to my concern form earlier comments: this test relies on exposing the internals of EmailNotifier. In other words, if not for this test, would createEmail need to be a non-private method? 

What I'd recommend doing instead is mocking out the Gmail service, capturing the sent email, and asserting that its from address is equal to "Adlib-Step@gmail.com"

Reply:
I am confused because send takes a message but form what I've seen the extraction methods (get) are most within the MimeMessage class ?